app/models/course/requirement.rb

```diff
@@ -0,0 +1,4 @@
+class Course::Requirement < ActiveRecord::Base
+  actable
+  belongs_to :has_requirement, polymorphic: true # naming is hard
```

Comment:
Maybe we should use standard computer science terminology. This is an abstraction condition. The things which have conditions are called conditionals. Things which implement this are also conditions.